kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/clients/records/KcatConsumerRecord.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.systemtests.clients.records;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * The type Kcat consumer record.
+ */
+@JsonIgnoreProperties(ignoreUnknown = true)
+public class KcatConsumerRecord extends ConsumerRecord {
+
+    /**
+     * Instantiates a new Kcat consumer record.
+     *
+     * @param headers the headers
+     * @param topic the topic
+     * @param key the key
+     * @param payload the payload
+     * @param partition the partition
+     * @param offset the offset
+     */
+    @JsonCreator
+    public KcatConsumerRecord(@JsonProperty("headers") List<String> headers,
+                              @JsonProperty("topic") String topic,
+                              @JsonProperty("key") String key,
+                              @JsonProperty("payload") String payload,
+                              @JsonProperty("partition") int partition,
+                              @JsonProperty("offset") long offset) {
+        this.recordHeaders = new HashMap<>();
+        if (headers != null) {
+            if (headers.size() % 2 != 0) {
```

Comment:
Depending on the List impl, size might be O(n), so in general, call it only once.  It won't matter here really because the lists are tiny, but I'm trying to promote good programming practice.